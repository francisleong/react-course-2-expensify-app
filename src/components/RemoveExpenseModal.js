import React from 'react';
import Modal from 'react-modal';
import moment from 'moment';
import numeral from 'numeral';

const RemoveExpenseModal = (props) => (
  <Modal
    isOpen={props.removeExpenseOption}
    contentLabel="Confirm Expense Removal"
    closeTimeoutMS={200}
    className="modal"
  >
    <h1>Are you sure you want to remove this expense?</h1>
    <h3>This can not be undone!</h3>
    <button
      className="button"
      onClick={(e) => {
        props.handleRemoveExpenseConfirmation(e);
        props.handleRemoveExpense(e);
      }}
    >
      Yes
    </button>
    <button
      className="button button--secondary"
      onClick={props.handleRemoveExpenseConfirmation}
    >
      No
    </button>
  </Modal>
);

export default RemoveExpenseModal;