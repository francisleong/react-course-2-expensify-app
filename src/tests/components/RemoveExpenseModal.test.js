import React from 'react';
import { shallow } from 'enzyme';
import RemoveExpenseModal from '../../components/RemoveExpenseModal';

let removeExpenseOption, handleRemoveExpenseConfirmation, handleRemoveExpense, wrapper;

beforeEach(() => {
  removeExpenseOption = undefined;
  handleRemoveExpenseConfirmation = jest.fn();
  handleRemoveExpense = jest.fn();
  wrapper = shallow(
    <RemoveExpenseModal
      removeExpenseOption={removeExpenseOption}
      handleRemoveExpenseConfirmation={handleRemoveExpenseConfirmation}
      handleRemoveExpense={handleRemoveExpense}
    />
  );

});

test('should render RemoveExpenseModal correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handleRemoveExpense (yes)', () => {
  wrapper.find('button').at(0).simulate('click');
  expect(handleRemoveExpenseConfirmation).toHaveBeenCalled();
  expect(handleRemoveExpense).toHaveBeenCalled();
});

test('should handleRemoveExpenseConfirmation (no)', () => {
  wrapper.find('button').at(1).simulate('click');
  expect(handleRemoveExpenseConfirmation).toHaveBeenCalled();
});