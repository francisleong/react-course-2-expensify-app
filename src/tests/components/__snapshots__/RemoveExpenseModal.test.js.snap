// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RemoveExpenseModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirm Expense Removal"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <h1>
    Are you sure you want to remove this expense?
  </h1>
  <h3>
    This can not be undone!
  </h3>
  <button
    className="button"
    onClick={[Function]}
  >
    Yes
  </button>
  <button
    className="button button--secondary"
    onClick={[Function]}
  >
    No
  </button>
</Modal>
`;
